hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java

```diff
@@ -138,22 +154,27 @@ public HTSPrecompiledContract(
 			final GlobalDynamicProperties dynamicProperties,
 			final GasCalculator gasCalculator,
 			final AccountRecordsHistorian recordsHistorian,
-			final SoliditySigsVerifier sigsVerifier,
+			final TxnAwareSoliditySigsVerifier sigsVerifier,
 			final DecodingFacade decoder,
 			final SyntheticTxnFactory syntheticTxnFactory,
-			final EntityCreator creator,
-			final DissociationFactory dissociationFactory
+			final ExpiringCreations creator,
+			final DissociationFactory dissociationFactory,
+			final EntityIdSource ids,
```

Comment:
Since precompiles cannot safely create new entities at this time, I'd lean toward _not_ injecting the `EntityIdSource` and always passing [the `NOOP_ID_SOURCE`](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/ledger/ids/ExceptionalEntityIdSource.java#L30) to the `HederaTokenStore` constructor.